cli/src/pcluster/config/cluster_config.py

```diff
@@ -1103,14 +1130,15 @@ def vpc_id(self):
         return AWSApi.instance().ec2.get_subnet_vpc(self.head_node.networking.subnet_id)
 
     @property
-    def ami_id(self):
-        """Get the image id of the cluster."""
+    def headnode_ami_id(self):
+        """Get the image id of the HeadNode."""
         ami_filters = self.dev_settings.ami_search_filters if self.dev_settings else None
-        return (
-            self.image.custom_ami
-            if self.image.custom_ami
-            else AWSApi.instance().ec2.get_official_image_id(self.image.os, self.head_node.architecture, ami_filters)
-        )
+        if self.head_node.image and self.head_node.image.custom_ami:
+            return self.head_node.image.custom_ami
+        elif self.image.custom_ami:
+            return self.image.custom_ami
+        else:
+            return AWSApi.instance().ec2.get_official_image_id(self.image.os, self.head_node.architecture, ami_filters)
```

Comment:
we should use the official_ami you defined below.

Reply:
done